vscode/src/azure/commands.ts

```diff
@@ -21,13 +22,27 @@ import { QuantumUris, compileToBitcode } from "./networkRequests";
 import { getQirForActiveWindow } from "../qirGeneration";
 
 const corsDocsUri = "https://github.com/microsoft/qsharp/wiki/Enabling-CORS";
+const workspacesSecret = "qsharp-vscode.workspaces";
 
-export function initAzureWorkspaces() {
+export async function initAzureWorkspaces(context: vscode.ExtensionContext) {
   const workspaceTreeProvider = new WorkspaceTreeProvider();
   vscode.window.createTreeView("quantum-workspaces", {
     treeDataProvider: workspaceTreeProvider,
   });
 
+  // Add any previously saved workspaces
+  const savedWorkspaces = await context.secrets.get(workspacesSecret);
+  if (savedWorkspaces) {
+    log.debug("Loading workspaces: ", savedWorkspaces);
+    const workspaces: WorkspaceConnection[] = JSON.parse(savedWorkspaces);
+    for (const workspace of workspaces) {
```

Comment:
Should we use `Promise.all()`?

Reply:
I actually re-wrote this a couple times after hitting some async hazards, and this was the safest and simplest code. There are some subtle bugs and gotchas when you iterate through a list and kick of async work for each item. (I expect the vast majority of users will only have the one workspace in the tree anyway).

If you have a strong opinion though, I could be convinced.